echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/pipelinetriggers/monitor/TriggerMonitor.java

```diff
@@ -97,41 +86,36 @@ protected void onEchoResponse(final TriggerEvent event) {
     registry.gauge("echo.events.per.poll", 1);
   }
 
-  protected Action1<TriggerMatchParameters> triggerEachMatch() {
-    return parameters -> {
-      triggerEachMatchFrom(parameters.pipelines).call(parameters.event);
-    };
-  }
-
-  protected Action1<TriggerEvent> triggerEachMatchFrom(final List<Pipeline> pipelines) {
-    return event -> {
-      if (isSuccessfulTriggerEvent(event)) {
-        Observable.from(pipelines)
-          .doOnCompleted(() -> onEventProcessed(event))
-          .map(withMatchingTrigger(event))
-          .filter(Optional::isPresent)
-          .map(Optional::get)
-          .doOnNext(this::onMatchingPipeline)
-          .subscribe(subscriber, this::onSubscriberError);
-      } else {
-        onEventProcessed(event);
-      }
-    };
+  protected void triggerEachMatchFrom(final TriggerEvent event, List<Pipeline> pipelines) {
```

Comment:
Given that we're refactoring this -- can we rename this to something clearer? Maybe `triggerEachMatchingPipeline`?

Reply:
Good idea, done